pandas/core/reshape/merge.py

```diff
@@ -814,12 +814,12 @@ def _get_merge_keys(self):
         join_names = []
         right_drop = []
         left_drop = []
+
         left, right = self.left, self.right
+        list_types = (np.ndarray, Series, Index)
 
-        is_lkey = lambda x: isinstance(
-            x, (np.ndarray, Series)) and len(x) == len(left)
-        is_rkey = lambda x: isinstance(
-            x, (np.ndarray, Series)) and len(x) == len(right)
+        is_lkey = lambda x: isinstance(x, list_types) and len(x) == len(left)
```

Comment:
can you use ``is_list_like`` here?

Reply:
I thought about that, but passing in `list` or `tuple` is not accepted (later on, we check for the `dtype` attribute.

Reply:
maybe ``lambda x: is_list_like(x) and hasattr(x, 'dtype')``?

just trying to avoid hardcoding (actually you could add that as ``is_array_like`` in ``dtypes/common``, seems useful

Reply:
Yeah, `is_array_like` makes sense as an abstraction.  I'll add that.